helix-core/src/main/java/org/apache/helix/controller/rebalancer/strategy/CrushRebalanceStrategy.java

```diff
@@ -89,7 +89,15 @@ public ZNRecord computePartitionAssignment(final List<String> allNodes,
       long data = partitionName.hashCode();
 
       // apply the placement rules
-      List<Node> selected = select(topNode, data, _replicas, eventId);
+      List<Node> selected;
+      try {
+        selected = select(topNode, data, _replicas, eventId);
+      } catch (IllegalStateException e) {
+        String errorMessage = String
```

Comment:
Did you use helix format template?

Reply:
@dasahcc I just changed my template and updated my commit. Thanks.